pandas/core/internals/blocks.py

```diff
@@ -596,6 +597,17 @@ def astype(self, dtype, copy: bool = False, errors: str = "raise"):
 
             return self.make_block(Categorical(self.values, dtype=dtype))
 
+        elif (  # GH8628
+            is_categorical_dtype(self.values.dtype)
+            and not (is_object_dtype(dtype) or is_string_like_dtype(dtype))
```

Comment:
Could define a new method for this in `core/dtypes/common.py` (`is_string_like_or_object_dtype`) ?